### ISSUE TITLE
Results for Iris dataset

### ISSUE THREAD

**Issue by Bolu98** [tusharsarkar3/XBNet#5]
Hello,

I am trying to recreate your results for the Iris dataset. I have installed XBNET and all the required libraries. 

I am following the example code provided in the READ ME and I am having some trouble with running the code. 

Could you please let me know what input and output dimensions you used for layers 1 and 2, with their respective biases, as well as what your last layer was?

Thank you very much.

**Comment by tusharsarkar3** [comment]
layer 1 input: 5
layer 1 output: 4
Bias: 1 (We just have to provide true or false for this so 1 will show it's true)
layer 2 input: 4
layer 2 output: 3
Bias: 1 (We just have to provide true or false for this so 1 will show it's true)
Then select 3 which stands for None and you are good to go.
I'm sorry for the late response. Hope this helps you!

**Comment by Bolu98** [comment]
Hi,

Thank you very much for getting back to me with this information.

When I try to run this on my Mac, I get a zsh: sementation fault error from the line `model = XBNETClassifier(X_train,y_train,2)`. 

My package versions are given below: 

Python: 3.9.5
torch: 1.9.0
numpy: 1.20.2
pandas: 1.2.5
scikit-learn: 0.24.2
XBNet: 1.3.1
xgboost: 1.4.2

Do you have any ideas for why this is happening?

**Comment by tusharsarkar3** [comment]
No, I'm not really sure why this is happening exactly. Try creating a new virtual environment and try installing XBNet in that environment. Check out this [link](https://www.kaggle.com/gauravduttakiit/xbnet-extremely-boosted-nn-on-iris-dataset) to see how XBNet works on the Kaggle notebook and try to implement it there or google colab. Also please attach the screenshot of the entire error message so that I can get to the crux of this issue and try to solve it.

**Comment by Bolu98** [comment]
Hello,

Thank you very much for all your help so far.

When I use Google colab to run the code given in the Kaggle notebook, everything works and the output is very similar to yours. 

However, I created a new virtual environment and installed XBNet there and tried to run the code in your example READ ME and I get a segmentation fault error. I have attached the screenshot showing my error message.

![Screenshot 2021-07-12 at 16 31 18](https://user-images.githubusercontent.com/34072762/125315289-de268800-e32e-11eb-942b-b58b47a64960.png)

**Comment by tusharsarkar3** [comment]
Happy to help!
Unfortunately, I have no idea about that segmentation fault so I'll label this issue so that someone familiar with such an issue can help you out

**Comment by athzPadilla** [comment]
Hello,  sorry for use this thread but I have issues with Iris data set too :(

I have installed XBNet and all the required libraries, besides I used the number of input/output layers that you mentioned for this dataset, but  when I ran:
 _"run_XBNET(X_train,X_test,y_train,y_test,model,criterion,optimizer,32,300)"_

I got:  _TypeError: min(): argument 'input' (position 1) must be Tensor, not NoneType_ like this 
![image](https://github.com/tusharsarkar3/XBNet/assets/100897429/8e33bd65-e6cd-46a5-9b38-a675081dc212)

Do you have any idea why a type error?
